wait/wait_test.go

```diff
@@ -0,0 +1,46 @@
+package wait
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/go-connections/nat"
+	tcexec "github.com/testcontainers/testcontainers-go/exec"
+)
+
+var ErrPortNotFound = errors.New("port not found")
+
+type MockStrategyTarget struct {
+	HostImpl       func(context.Context) (string, error)
+	PortsImpl      func(context.Context) (nat.PortMap, error)
+	MappedPortImpl func(context.Context, nat.Port) (nat.Port, error)
+	LogsImpl       func(context.Context) (io.ReadCloser, error)
+	ExecImpl       func(context.Context, []string, ...tcexec.ProcessOption) (int, io.Reader, error)
```

Comment:
I'm checking that `ExecImpl` is not used anywhere else but in https://github.com/testcontainers/testcontainers-go/pull/944/files#diff-01e1c8168e5c89a7f968caae722f60ee322b183eddf9169560ba0ccef4abf0bdR41.

Do you think we need it, or just having a NOOP is enough for the mock strategy executing commands?

Reply:
Oh oh I'm sorry 🙏 you're right, the diff was not open. I'm resolving this comment now